From 903883ba09721833f1c94d98a0a246ca4e671242 Mon Sep 17 00:00:00 2001
From: Larry Finger <Larry.Finger@lwfinger.net>
Date: Thu, 16 May 2013 20:03:57 -0500
Subject: [PATCH 161/210] rtl8192du: Rework debugging

The changes include creating a module parameter to set the debug level
at module load time, and eliminating any debug configuration variables.

Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>
---
 core/rtw_debug.c    |  66 ++++++++++++++--------------
 core/rtw_mlme_ext.c |  25 +++++------
 core/rtw_xmit.c     |   2 +-
 include/autoconf.h  |   5 ---
 include/rtw_debug.h | 124 ++++++++++++++++++++++++----------------------------
 os_dep/os_intfs.c   |   6 ++-
 os_dep/recv_linux.c |  19 ++++----
 7 files changed, 113 insertions(+), 134 deletions(-)

--- a/drivers/staging/rtl8192du/core/rtw_debug.c
+++ b/drivers/staging/rtl8192du/core/rtw_debug.c
@@ -22,40 +22,38 @@
 
 #include <rtw_debug.h>
 
-/* ifdef CONFIG_DEBUG_RTL871X */
+u32 GlobalDebugLevel = _drv_err_;
 
-	u32 GlobalDebugLevel = _drv_err_;
-
-	u64 GlobalDebugComponents = 
-			_module_rtl871x_xmit_c_ |
-			_module_xmit_osdep_c_ |
-			_module_rtl871x_recv_c_ |
-			_module_recv_osdep_c_ |
-			_module_rtl871x_mlme_c_ |
-			_module_mlme_osdep_c_ |
-			_module_rtl871x_sta_mgt_c_ |
-			_module_rtl871x_cmd_c_ |
-			_module_cmd_osdep_c_ |
-			_module_rtl871x_io_c_ |
-			_module_io_osdep_c_ |
-			_module_os_intfs_c_|
-			_module_rtl871x_security_c_|
-			_module_rtl871x_eeprom_c_|
-			_module_hal_init_c_|
-			_module_hci_hal_init_c_|
-			_module_rtl871x_ioctl_c_|
-			_module_rtl871x_ioctl_set_c_|
-			_module_rtl871x_ioctl_query_c_|
-			_module_rtl871x_pwrctrl_c_|
-			_module_hci_intfs_c_|
-			_module_hci_ops_c_|
-			_module_hci_ops_os_c_|
-			_module_rtl871x_ioctl_os_c|
-			_module_rtl8712_cmd_c_|
-			_module_hal_xmit_c_|
-			_module_rtl8712_recv_c_ |
-			_module_mp_ |
-			_module_efuse_;
+u64 GlobalDebugComponents = 
+	_module_rtl871x_xmit_c_ |
+	_module_xmit_osdep_c_ |
+	_module_rtl871x_recv_c_ |
+	_module_recv_osdep_c_ |
+	_module_rtl871x_mlme_c_ |
+	_module_mlme_osdep_c_ |
+	_module_rtl871x_sta_mgt_c_ |
+	_module_rtl871x_cmd_c_ |
+	_module_cmd_osdep_c_ |
+	_module_rtl871x_io_c_ |
+	_module_io_osdep_c_ |
+	_module_os_intfs_c_|
+	_module_rtl871x_security_c_|
+	_module_rtl871x_eeprom_c_|
+	_module_hal_init_c_|
+	_module_hci_hal_init_c_|
+	_module_rtl871x_ioctl_c_|
+	_module_rtl871x_ioctl_set_c_|
+	_module_rtl871x_ioctl_query_c_|
+	_module_rtl871x_pwrctrl_c_|
+	_module_hci_intfs_c_|
+	_module_hci_ops_c_|
+	_module_hci_ops_os_c_|
+	_module_rtl871x_ioctl_os_c|
+	_module_rtl8712_cmd_c_|
+	_module_hal_xmit_c_|
+	_module_rtl8712_recv_c_ |
+	_module_mp_ |
+	_module_efuse_;
 
 /* endif */
 
--- a/drivers/staging/rtl8192du/core/rtw_mlme_ext.c
+++ b/drivers/staging/rtl8192du/core/rtw_mlme_ext.c
@@ -9484,28 +9484,25 @@ static void process_80211d(struct rtw_ad
 		}
 		chplan_ap.Len = i;
 
-#ifdef CONFIG_DEBUG_RTL871X
 		i = 0;
-		printk("%s: AP[%s] channel plan {", __func__, bssid->Ssid.Ssid);
+		DBG_8192D("%s: AP[%s] channel plan {", __func__,
+			  bssid->Ssid.Ssid);
 		while ((i < chplan_ap.Len) && (chplan_ap.Channel[i] != 0)) {
-			printk("%02d,", chplan_ap.Channel[i]);
+			DBG_8192D("%02d,", chplan_ap.Channel[i]);
 			i++;
 		}
-		printk("}\n");
-#endif
+		DBG_8192D("}\n");
 
 		memcpy(chplan_sta, pmlmeext->channel_set, sizeof(chplan_sta));
-#ifdef CONFIG_DEBUG_RTL871X
 		i = 0;
-		printk("%s: STA channel plan {", __func__);
+		DBG_8192D("%s: STA channel plan {", __func__);
 		while ((i < MAX_CHANNEL_NUM) && (chplan_sta[i].ChannelNum != 0)) {
-			printk("%02d(%c),", chplan_sta[i].ChannelNum,
+			DBG_8192D("%02d(%c),", chplan_sta[i].ChannelNum,
 			       chplan_sta[i].ScanType ==
 			       SCAN_PASSIVE ? 'p' : 'a');
 			i++;
 		}
-		printk("}\n");
-#endif
+		DBG_8192D("}\n");
 
 		memset(pmlmeext->channel_set, 0, sizeof(pmlmeext->channel_set));
 		chplan_new = pmlmeext->channel_set;
@@ -9650,17 +9647,15 @@ static void process_80211d(struct rtw_ad
 
 		pmlmeext->update_channel_plan_by_ap_done = 1;
 
-#ifdef CONFIG_DEBUG_RTL871X
 		k = 0;
-		printk("%s: new STA channel plan {", __func__);
+		DBG_8192D("%s: new STA channel plan {", __func__);
 		while ((k < MAX_CHANNEL_NUM) && (chplan_new[k].ChannelNum != 0)) {
-			pr_info("%02d(%c),", chplan_new[k].ChannelNum,
+			DBG_8192D("%02d(%c),", chplan_new[k].ChannelNum,
 			        chplan_new[k].ScanType ==
 			        SCAN_PASSIVE ? 'p' : 'c');
 			k++;
 		}
-		printk("}\n");
-#endif
+		DBG_8192D("}\n");
 	}
 
 	/*  If channel is used by AP, set channel scan type to active */
--- a/drivers/staging/rtl8192du/core/rtw_xmit.c
+++ b/drivers/staging/rtl8192du/core/rtw_xmit.c
@@ -770,7 +770,7 @@ _func_enter_;
 				 ("xmitframe_addmic: pattrib->last_txcmdsz =%d!!!\n",
 				 pattrib->last_txcmdsz));
 			RT_TRACE(_module_rtl871x_xmit_c_, _drv_err_,
-				 ("xmitframe_addmic: mic[0]= 0x%.2x , mic[1]= 0x%.2x , mic[2]= 0x%.2x , mic[3]= 0x%.2x\n\mic[4]= 0x%.2x , mic[5]= 0x%.2x , mic[6]= 0x%.2x , mic[7]= 0x%.2x !!!!\n",
+				 ("xmitframe_addmic: mic[0]= 0x%.2x , mic[1]= 0x%.2x , mic[2]= 0x%.2x , mic[3]= 0x%.2x\nmic[4]= 0x%.2x , mic[5]= 0x%.2x , mic[6]= 0x%.2x , mic[7]= 0x%.2x !!!!\n",
 				 mic[0], mic[1], mic[2], mic[3], mic[4], mic[5],
 				 mic[6], mic[7]));
 			/* add mic code  and add the mic code length in last_txcmdsz */
--- a/drivers/staging/rtl8192du/include/autoconf.h
+++ b/drivers/staging/rtl8192du/include/autoconf.h
@@ -178,11 +178,6 @@
 
 #define MP_DRIVER 0
 
-/*
- * Debug  Related Config
- */
-/* define CONFIG_DEBUG_RTL871X */
-
 #define DBG 0
 
 #define CONFIG_DEBUG_RTL819X
--- a/drivers/staging/rtl8192du/include/rtw_debug.h
+++ b/drivers/staging/rtl8192du/include/rtw_debug.h
@@ -28,13 +28,11 @@
 #define	_no_debug_			0
 #define _drv_emerg_			1
 #define _drv_alert_			2
-#define _drv_crit_			3
-#define _drv_err_			4
-#define	_drv_warning_		5
-#define _drv_notice_			6
-#define _drv_info_			7
-#define _drv_dump_			8
-#define	_drv_debug_		9
+#define _drv_err_			3
+#define	_drv_warning_			4
+#define _drv_notice_			5
+#define _drv_info_			6
+#define	_drv_debug_			7
 
 
 #define _module_rtl871x_xmit_c_		BIT(0)
@@ -158,84 +156,74 @@
 
 #undef	_dbgdump
 
-#ifdef CONFIG_DEBUG_RTL871X
-
 #ifndef _RTL871X_DEBUG_C_
 	extern u32 GlobalDebugLevel;
 	extern u64 GlobalDebugComponents;
 #endif
 
-#define _dbgdump	printk
-
-#endif /* CONFIG_DEBUG_RTL871X */
-
-
-#if	defined (_dbgdump) && defined (_MODULE_DEFINE_)
-
-		#undef RT_TRACE
-		#define RT_TRACE(_Comp, _Level, Fmt)\
-		do {\
-			if ((_Comp & GlobalDebugComponents) && (_Level <= GlobalDebugLevel)) {\
-				_dbgdump("%s [0x%08x,%d]", RTL871X_MODULE_NAME, (unsigned int)_Comp, _Level);\
-				_dbgdump Fmt;\
-			}\
-		}while (0)
+#define _dbgdump	pr_info
 
+#if defined (_dbgdump) && defined (_MODULE_DEFINE_)
+	#undef RT_TRACE
+	#define RT_TRACE(_Comp, _Level, Fmt)				\
+	do {								\
+		if (_Level <= GlobalDebugLevel) {			\
+			_dbgdump("%s [0x%08x,%d]", RTL871X_MODULE_NAME,	\
+				 (unsigned int)_Comp, _Level);		\
+			_dbgdump Fmt;					\
+		}							\
+	} while (0)
 #endif
 
+#if defined (_dbgdump)
+	#undef  _func_enter_
+	#define _func_enter_						\
+	do {								\
+		if (GlobalDebugLevel >= _drv_debug_) {			\
+			_dbgdump("\n %s : %s enters at %d\n",		\
+				 RTL871X_MODULE_NAME, __func__, __LINE__);\
+		}						\
+	} while (0)
 
-#if	defined (_dbgdump)
+	#undef  _func_exit_
+	#define _func_exit_						\
+	do {								\
+		if (GlobalDebugLevel >= _drv_debug_) {			\
+			_dbgdump("\n %s : %s exits at %d\n",		\
+				 RTL871X_MODULE_NAME, __func__, __LINE__); \
+		}							\
+	} while (0)
 
-		#undef  _func_enter_
-		#define _func_enter_ \
-		do {	\
-			if (GlobalDebugLevel >= _drv_debug_) \
-			{																	\
-				_dbgdump("\n %s : %s enters at %d\n", RTL871X_MODULE_NAME, __func__, __LINE__);\
-			}		\
-		} while (0)
-
-		#undef  _func_exit_
-		#define _func_exit_ \
-		do {	\
-			if (GlobalDebugLevel >= _drv_debug_) \
-			{																	\
-				_dbgdump("\n %s : %s exits at %d\n", RTL871X_MODULE_NAME, __func__, __LINE__); \
-			}	\
-		} while (0)
-
-		#undef RT_PRINT_DATA
-		#define RT_PRINT_DATA(_Comp, _Level, _TitleString, _HexData, _HexDataLen)			\
-			if (((_Comp) & GlobalDebugComponents) && (_Level <= GlobalDebugLevel))	\
-			{									\
-				int __i;								\
-				u8	*ptr = (u8 *)_HexData;				\
-				_dbgdump("Rtl871x: ");						\
-				_dbgdump(_TitleString);						\
-				for (__i=0; __i<(int)_HexDataLen; __i++)				\
-				{								\
-					_dbgdump("%02X%s", ptr[__i], (((__i + 1) % 4) == 0)?"  ":" ");	\
-					if (((__i + 1) % 16) == 0)	_dbgdump("\n");			\
-				}								\
-				_dbgdump("\n");							\
+	#undef RT_PRINT_DATA
+	#define RT_PRINT_DATA(_Comp, _Level, _TitleString, _HexData,	\
+			      _HexDataLen)				\
+		if (((_Comp) & GlobalDebugComponents) &&		\
+		    (_Level <= GlobalDebugLevel)) {			\
+			int __i;					\
+			u8 *ptr = (u8 *)_HexData;			\
+			_dbgdump("Rtl871x: ");				\
+				_dbgdump(_TitleString);			\
+				for (__i = 0; __i < (int)_HexDataLen; __i++) {\
+					_dbgdump("%02X%s", ptr[__i],	\
+						 (((__i + 1) % 4) == 0) ?\
+						 " " : " ");	\
+					if (((__i + 1) % 16) == 0)	\
+						_dbgdump("\n");		\
+				}					\
+				_dbgdump("\n");				\
 			}
 #endif
 
-
-#ifdef CONFIG_DEBUG_RTL819X
-
 #undef	_dbgdump
 
-#define _dbgdump	printk
-
-#endif /* CONFIG_DEBUG_RTL819X */
+#define _dbgdump	pr_info
 
 extern u32 GlobalDebugLevel;
 #define LOG_LEVEL(level, ...)\
 	do {\
-		if (level <= GlobalDebugLevel) {\
-			pr_debug(__VA_ARGS__);\
-		}\
+		if (level <= GlobalDebugLevel) {			\
+			pr_debug(__VA_ARGS__);				\
+		}							\
 	} while (0)
 
 #define DBG_8192D(...) LOG_LEVEL(_drv_debug_ ,  __VA_ARGS__)
--- a/drivers/staging/rtl8192du/os_dep/os_intfs.c
+++ b/drivers/staging/rtl8192du/os_dep/os_intfs.c
@@ -55,7 +55,8 @@ static int rtw_frag_thresh = 2346;/*  */
 static int rtw_preamble = PREAMBLE_LONG;/* long, short, auto */
 static int rtw_scan_mode = 1;/* active, passive */
 static int rtw_adhoc_tx_pwr = 1;
-static int rtw_soft_ap = 0;
+static int rtw_soft_ap;
+static int debug = 1;
 /* int smart_ps = 1; */
 #ifdef CONFIG_POWER_SAVING
 static int rtw_power_mgnt = 1;
@@ -69,7 +70,9 @@ int rtw_power_mgnt = PS_MODE_ACTIVE;
 int rtw_ips_mode = IPS_NONE;
 #endif
 module_param(rtw_ips_mode, int, 0644);
+module_param(debug, int, 0644);
 MODULE_PARM_DESC(rtw_ips_mode,"The default IPS mode");
+MODULE_PARM_DESC(debug, "Set debug level (0-7) (default 1)");
 
 static int rtw_radio_enable = 1;
 static int rtw_long_retry_lmt = 7;
@@ -759,6 +762,7 @@ _func_enter_;
 #endif
 
 	registry_par->mac_phy_mode = rtw_mac_phy_mode;
+	GlobalDebugLevel = debug;
 
 #ifdef CONFIG_80211D
 	registry_par->enable80211d = (u8)rtw_80211d;
--- a/drivers/staging/rtl8192du/os_dep/recv_linux.c
+++ b/drivers/staging/rtl8192du/os_dep/recv_linux.c
@@ -259,28 +259,27 @@ _func_enter_;
 
 	skb->len = precv_frame->u.hdr.len;
 
-	RT_TRACE(_module_recv_osdep_c_,_drv_info_,("\n skb->head=%p skb->data=%p skb->tail=%p skb->end=%p skb->len=%d\n", skb->head, skb->data, skb->tail, skb->end, skb->len));
+	RT_TRACE(_module_recv_osdep_c_, _drv_info_,
+		 ("\n skb->head=%p skb->data=%p skb->tail=%p skb->end=%p skb->len=%d\n",
+		 skb->head, skb->data, skb_tail_pointer(skb),
+		 skb_end_pointer(skb), skb->len));
 
-	if (check_fwstate(pmlmepriv, WIFI_AP_STATE) == true)
-	{
-		struct sk_buff *pskb2=NULL;
+	if (check_fwstate(pmlmepriv, WIFI_AP_STATE) == true) {
+		struct sk_buff *pskb2 = NULL;
 		struct sta_info *psta = NULL;
 		struct sta_priv *pstapriv = &padapter->stapriv;
 		struct rx_pkt_attrib *pattrib = &precv_frame->u.hdr.attrib;
 		int bmcast = IS_MCAST(pattrib->dst);
 
-		if (_rtw_memcmp(pattrib->dst, myid(&padapter->eeprompriv), ETH_ALEN)==false)
-		{
-			if (bmcast)
-			{
+		if (_rtw_memcmp(pattrib->dst, myid(&padapter->eeprompriv), ETH_ALEN)==false) {
+			if (bmcast) {
 				psta = rtw_get_bcmc_stainfo(padapter);
 				pskb2 = skb_clone(skb, GFP_ATOMIC);
 			} else {
 				psta = rtw_get_stainfo(pstapriv, pattrib->dst);
 			}
 
-			if (psta)
-			{
+			if (psta) {
 				struct net_device *pnetdev= (struct net_device*)padapter->pnetdev;
 
 				skb->dev = pnetdev;
